keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,966 @@
+* gitRepo volume types can be exploited to gain remote code execution as root on the nodes as shown in [CVE-2024-10220](https://nvd.nist.gov/vuln/detail/cve-2024-10220)
+* gitRepo has perfectly workable alternatives:
+  * Using [git-sync](https://github.com/kubernetes/git-sync)
+  * Using an initContainer. See https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo for more details
+* gitRepo has been deprecated for a long time and is unmaintained
+* gitRepo has low usage (based on limited data)
+* there exists precedent for removing volume drivers which were unmaintained like [glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+
+Given this, we think kubernetes should EOL this feature with urgency.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Remove a vector for remote code execution through an ancient and unmaintained
+volume plugin.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Remove in-tree gitRepo volume support.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Kubernetes drops support for gitRepo volume types by removing the in-tree driver
+that actually mounts the volume in the pod. In this case we will still keep the
+gitRepo volume in the API. This removal is similar to removal of support for
+[glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+which was deprecated in v1.25 and removed in v1.26.
+
+Sig-storage was actually very close to doing this and had even added a notice of
+removal in the release notes for v1.32. But due to concerns raised by sig-arch
+they rolled back the notice. sig-storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The biggest risk is in breaking users. Despite being deprecated for years, there
+may be some users who are still using it and their workloads would be broken 
+when their clusters are upgraded. We plan to mitigate the risk by allowing
+users to opt-in to re-enabling the driver while they fix the workloads.
+
```

Comment:
- How about defining a number of minor releases for which the support can be re-enabled?
- Do we want to warn users when their PodTemplate or Pod uses this volume type
- Will we do any communications to end users? (hint: “no“ isn't going to fly)

Reply:
Add some more details. Thanks!